app/components/signum/notification_body/component.html.slim

```diff
@@ -30,12 +30,12 @@
             | Close
           svg.h-5.w-5[xmlns="http://www.w3.org/2000/svg" viewbox="0 0 20 20" fill="currentColor" aria-hidden="true"]
             path[d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z"]
-    - if signal.metadata.present? && signal.metadata["buttons"].present?
+    - if signal.buttons.present?
       .signum-notification-body__mb__bmc
-        - signal.metadata["buttons"].each do | button |
+        - signal.buttons.each do | button |
           - next unless button.is_a?(Hash)
           .signum-notification-body__mb__bmc__bc
-            a.signum-notification-body__mb__bmc__bc__b*{href: button.fetch('url', '#'), target:button.fetch("target", "_blank")}.merge(button.fetch("link_attributes", {}))
+            a.signum-notification-body__mb__bmc__bc__b*{href: button.fetch('url', '#'), target:button.fetch("target", "_blank"), 'data-action':"click->signum-notification-body#closeNotificationDrawer"}.merge(button.fetch("link_attributes", {}))
```

Comment:
same as above, check button[:close], if set to true, close the notification, not the drawer.